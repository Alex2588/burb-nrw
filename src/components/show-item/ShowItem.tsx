import React from 'react';
import { Link } from 'react-router-dom';

import { ShowItemProps } from 'shared/types/custom/show-item-props';
import './show-item.styles.scss';

const ShowItem = ({ show }: ShowItemProps) => (
    <Link to={`/shows/${show.id}`}>
        <div className="show-item">
            <img className='image' style={{
                backgroundImage: `url(${show.image.medium})`
            }}
            alt=""
        />
        <div className='show-item-footer'>
            <span className='name'>{show.name}</span>
        </div>
        </div>
    </Link>
);

export default ShowItem;