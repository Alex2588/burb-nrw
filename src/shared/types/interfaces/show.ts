export interface Show {
    id: string,
    url: string,
    name: string,
    type: string,
    language: string,
    genres: string[],
    status: string,
    runtime: number,
    premiered: string,
    officialSite: string,
    schedule: { time: string, days: string[] },
    rating: { average: number },
    weight: number,
    network: {
        id: number,
        name: string,
        country: {
            name: string,
            code: string,
            timezone: string
        }
    },
    externals: {
        tvrage: number,
        thetvdb: number,
        imdb: string
    },
    image: {
        medium: string,
        original: string
    },
    summary: string,
    updated: string,
    _links: {
        self: {
            href: string
        },
        previousepisode: {
            href: string
        }
    }
}