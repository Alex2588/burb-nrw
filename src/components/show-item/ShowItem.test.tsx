import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { mock } from 'jest-mock-extended';

import ShowItem from './ShowItem';
import { Show } from 'shared/types/interfaces/show';

test('renders show-item component', async () => {
    const show = mock<Show>();
    show.name = 'Amazing show';

    const { container } = render(<ShowItem show={show} />, { wrapper: MemoryRouter });

    expect(screen.getByText(/Amazing show/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
});

