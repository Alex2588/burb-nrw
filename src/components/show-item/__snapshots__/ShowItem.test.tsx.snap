// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders show-item component 1`] = `
<a
  href="/shows/function () {
        return fn.apply(this, arguments);
      }"
>
  <div
    class="show-item"
  >
    <img
      alt=""
      class="image"
      style="background-image: url(undefined);"
    />
    <div
      class="show-item-footer"
    >
      <span
        class="name"
      >
        Amazing show
      </span>
    </div>
  </div>
</a>
`;
